components/validator/challenger.go

```diff
@@ -692,23 +694,36 @@ func (c *Challenger) ProveFault(ctx context.Context, outputIndex *big.Int, skipS
 
 	// When asserter timeout, skip finding fault position since the same segments have been stored in colosseum.
 	position := common.Big0
-	blockNumber := challenge.SegStart.Uint64()
+	blockNumber := challenge.SegStart
 	if !skipSelectPosition {
-		segments := chal.NewSegments(challenge.SegStart.Uint64(), challenge.SegSize.Uint64(), challenge.Segments)
+		segments := chal.NewSegments(blockNumber.Uint64(), challenge.SegSize.Uint64(), challenge.Segments)
 		position, err = c.selectFaultPosition(ctx, segments)
 		if err != nil {
 			return nil, err
 		}
 
-		blockNumber = challenge.SegStart.Uint64() + position.Uint64()
+		blockNumber = new(big.Int).Add(blockNumber, position)
 	}
 
-	fetchResult, err := c.cfg.ProofFetcher.FetchProofAndPair(blockNumber + 1)
+	targetBlockNumber := new(big.Int).Add(blockNumber, common.Big1)
+	cCtx, cCancel := context.WithTimeout(ctx, c.cfg.NetworkTimeout)
+	defer cCancel()
+	trace, err := c.l2Client.GetBlockTraceByNumber(cCtx, targetBlockNumber)
 	if err != nil {
-		return nil, fmt.Errorf("%w: blockNumber: %d", err, blockNumber)
+		return nil, fmt.Errorf("failed to get block trace(block number: %d): %w", targetBlockNumber.Uint64(), err)
 	}
 
-	proof, err := c.PublicInputProof(ctx, blockNumber)
+	traceBz, err := json.Marshal(trace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshaling trace(block number: %d): %w", targetBlockNumber.Uint64(), err)
+	}
+
+	fetchResult, err := c.cfg.ProofFetcher.FetchProofAndPair(ctx, string(traceBz))
```

Comment:
While waiting for the response of the request to the fetcher, it seems that this validator will be unable to respond to other challenges. The situation where this causes a timeout seems to be a problem.